java-frontend/src/main/java/org/sonar/java/collections/AVLTree.java

```diff
@@ -108,86 +91,48 @@ public V get(K key) {
   }
 
   @Override
-  public void forEach(PSet.Consumer<K> action) {
+  public void forEach(Consumer<K> action) {
     forEach(this, action);
   }
 
   @SuppressWarnings("unchecked")
-  private void forEach(AVLTree t, PSet.Consumer<K> action) {
+  private void forEach(AVLTree t, Consumer<K> action) {
     if (t.isEmpty()) {
       return;
     }
     forEach(t.left(), action);
-    action.accept((K) t.key());
     forEach(t.right(), action);
+    while (t != null) {
+      action.accept((K) t.key());
+      t = t.nextInBucket();
+    }
   }
 
   @Override
-  public void forEach(PMap.Consumer<K, V> consumer) {
+  public void forEach(BiConsumer<K, V> consumer) {
     forEach(this, consumer);
   }
 
   @SuppressWarnings("unchecked")
-  private void forEach(AVLTree t, PMap.Consumer<K, V> action) {
+  private void forEach(AVLTree t, BiConsumer<K, V> action) {
     if (t.isEmpty()) {
       return;
     }
     forEach(t.left(), action);
-    action.accept((K) t.key(), (V) t.value());
     forEach(t.right(), action);
-  }
-
-  @Override
-  public Iterator<Map.Entry<K, V>> entriesIterator() {
-    return new NodeIterator(this);
-  }
-
-  private class NodeIterator implements Iterator<Map.Entry<K, V>> {
-    private Deque<AVLTree> stack = new ArrayDeque<>();
-
-    public NodeIterator(AVLTree<K, V> node) {
-      descendLeft(node);
-    }
-
-    @Override
-    public boolean hasNext() {
-      return !stack.isEmpty();
-    }
-
-    private void descendLeft(AVLTree node) {
-      if (node != EMPTY) {
-        stack.addLast(node);
-        descendLeft(node.left());
-      }
-    }
-
-    @Override
-    public Map.Entry<K, V> next() {
-      if (!hasNext()) {
-        throw new NoSuchElementException();
-      }
-      AVLTree node = stack.removeLast();
-      if (node.right() != EMPTY) {
-        descendLeft(node.right());
-      }
-      return new AbstractMap.SimpleImmutableEntry<>((K) node.key(), (V) node.value());
+    while (t != null) {
+      action.accept((K) t.key(), (V) t.value());
+      t = t.nextInBucket();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Introduce a new variable instead of reusing the parameter "t". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1226)